packages/testing/src/api/index.ts

```diff
@@ -43,39 +50,49 @@ export type AppOptions = {
   projectId?: string;
   auth: object;
 };
+/** Construct a FirebaseApp authenticated with options.auth. */
 export function initializeTestApp(options: AppOptions): FirebaseApp {
-  let appOptions: FirebaseOptions;
-  if ('databaseName' in options) {
+  const app = initializeApp(options.databaseName, options.projectId);
+  // hijacking INTERNAL.getToken to bypass FirebaseAuth and allows specifying of auth headers
+  const unsecuredJwt = createUnsecuredJwt(options.auth);
+  (app as any).INTERNAL.getToken = () =>
+    Promise.resolve({ accessToken: unsecuredJwt });
+  return app;
+}
+
+export type AdminAppOptions = {
+  databaseName?: string;
+  projectId?: string;
+};
+/** Construct a FirebaseApp authenticated as an admin user. */
+export function initializeAdminApp(options: AdminAppOptions): FirebaseApp {
+  const app = initializeApp(options.databaseName, options.projectId);
+  // hijacking INTERNAL.getToken to bypass FirebaseAuth and allows specifying of auth headers
+  (app as any).INTERNAL.getToken = () =>
+    Promise.resolve({ accessToken: ADMIN_TOKEN });
+  return app;
+}
```

Comment:
can you combine the logic for `initializeTestApp` and `initializeAdminApp` since they're essentially the same thing with a different accessToken